### PR TITLE
Update readme to have a more complete compose file

### PR DESCRIPTION
**Pull request by edbo**
I changed the docker-compose a bit so you can also see the ui and it includes a fix that allows for uploading to the cluster more easily.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,14 +74,35 @@ services:
     supervisor:
         image: 31z4/storm:1.0.2
         container_name: supervisor
-        command: storm supervisor
+        command: >
+            storm supervisor
+            -c storm.zookeeper.servers="[\"zookeeper\"]"
```
Comment (31z4) [inline README.md:79]: This is done by default if no config provided. Please see https://github.com/31z4/storm-docker/blob/master/1.0.2/docker-entrypoint.sh#L15-L16

File: README.md [hunk 2]
```diff
@@ -71,14 +74,35 @@ services:
...
+    ui:
+        image: 31z4/storm:1.0.2
+        container_name: ui
+        command: >
+            storm ui
+            -c storm.zookeeper.servers="[\"zookeeper\"]"
+            -c nimbus.seeds="[\"nimbus\"]"
+        depends_on:
+            - nimbus
+            - zookeeper
```
Comment (31z4) [inline README.md:99]: I don't think the UI depends on Zookeeper.

File: README.md [hunk 3]
```diff
@@ -71,14 +74,35 @@ services:
...
+        links:
+            - nimbus
+            - zookeeper
```
Comment (31z4) [inline README.md:102]: Linking with Zookeeper is unnecessary.

**Review by 31z4 — CHANGES_REQUESTED** [review]

**Comment by 31z4** [comment]
@edbo could you please clarify on this

> it includes a fix that allows for uploading to the cluster more easily

What exactly do you mean?

**Comment by edbo** [comment]
I wasn't able to upload a topology to the storm cluster without setting a `storm.local.hostname`

**Comment by 31z4** [comment]
@edbo could you please create an issue for this describing all the details? Or do it right here.

File: README.md (hunk 1 above)
Comment (edbo) [reply README.md:79]: Nice didn't spot that.

**Comment by edbo** [comment]
I get the error when using a command like tihs:

```
storm  jar target/file-1.0-SNAPSHOT-jar-with-dependencies.jar com.somecompany.TestTopology test
```

```
448  [main] INFO  o.a.s.StormSubmitter - Generated ZooKeeper secret payload for MD5-digest: -7340068425218594486:-5911778086654076904
501  [main] INFO  o.a.s.s.a.AuthUtils - Got AutoCreds []
Exception in thread "main" java.lang.RuntimeException: org.apache.storm.thrift.transport.TTransportException: java.net.UnknownHostException: ee3f313730d9
    at org.apache.storm.security.auth.TBackoffConnect.retryNext(TBackoffConnect.java:64)
    at org.apache.storm.security.auth.TBackoffConnect.doConnectWithRetry(TBackoffConnect.java:56)
    at org.apache.storm.security.auth.ThriftClient.reconnect(ThriftClient.java:99)
    at org.apache.storm.security.auth.ThriftClient.<init>(ThriftClient.java:69)
    at org.apache.storm.utils.NimbusClient.<init>(NimbusClient.java:106)
    at org.apache.storm.utils.NimbusClient.getConfiguredClientAs(NimbusClient.java:78)
    at org.apache.storm.StormSubmitter.topologyNameExists(StormSubmitter.java:371)
    at org.apache.storm.StormSubmitter.submitTopologyAs(StormSubmitter.java:233)
    at org.apache.storm.StormSubmitter.submitTopology(StormSubmitter.java:311)
    at org.apache.storm.StormSubmitter.submitTopology(StormSubmitter.java:157)
    at com.bitgamelabs.LiveTransactionTopology.main(LiveTransactionTopology.java:29)
Caused by: org.apache.storm.thrift.transport.TTransportException: java.net.UnknownHostException: ee3f313730d9
    at org.apache.storm.thrift.transport.TSocket.open(TSocket.java:226)
    at org.apache.storm.thrift.transport.TFramedTransport.open(TFramedTransport.java:81)
    at org.apache.storm.security.auth.SimpleTransportPlugin.connect(SimpleTransportPlugin.java:103)
    at org.apache.storm.security.auth.TBackoffConnect.doConnectWithRetry(TBackoffConnect.java:53)
    ... 9 more
Caused by: java.net.UnknownHostException: ee3f313730d9
    at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
    at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
    at java.net.Socket.connect(Socket.java:589)
    at org.apache.storm.thrift.transport.TSocket.open(TSocket.java:221)
```

Setting the hostname to nimbus gets it working. I'm about to tweak it a bit to see if I can drop the other bits.

**Comment by edbo** [comment]
I realised after your message about the config settings you were setting that I was running your new docker compose file against your old docker image that didn't have those values set. Was confusing the hell out of me.

Anyway, I'm closing this now in favour of two simpler ones that separate the changes (https://github.com/31z4/storm-docker/pull/6 & https://github.com/31z4/storm-docker/pull/7).

**Pull request closed without merging**